### PR TITLE
ActiveNet module

### PR DESCRIPTION
**Pull request by mattshoaf**
Make sure these boxes are checked before asking for review of your pull request - thank you!

## General checks
- [x] All coding styles are fulfilled and there are no any issues reported by CodeSniffer CI. <br/>CI code sniffer errors
- [x] All tests are running and there are no failed tests reported by CI. <br/>Behat test results
- [x] [Documentation](https://github.com/ymcatwincities/openy/tree/8.x-1.x/docs) has been updated according to PR changes.
- [x] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/>Steps for review
- [x] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md).
- [x] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/>drupal.org email
- [x] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

## Steps for review
Drupal.org Issue: https://www.drupal.org/project/openy/issues/2953623

This module creates API endpoints for ActiveNet. After enabling the module, you should be able to input the config into the admin and use the ActiveNetClient class.

1. Enter activenet creds at /admin/config/services/activenet 
2. In your theme.theme file or a custom module:

```
use Drupal\activenet\ActiveNetClient;
$client = new $ActiveNetClient();
$centers =  $ActiveNetClient->getCenters();
print_r($centers); //returns json response of all centers

```
All available methods:
```
getSites()
getActivities()
getFlexRegPrograms()
getMembershipPackages()
getMembershipCategories()
```

### CODE REVIEW

File: modules/custom/activenet/src/ActivenetClient.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Drupal\activenet;
+
+use GuzzleHttp\Client;
+// use Drupal\activenet\ActivenetClientFactory;
```
Comment (podarok) [inline modules/custom/activenet/src/ActivenetClient.php]: remove debug code, please

**Review by podarok — CHANGES_REQUESTED** [review]

File: modules/custom/activenet/activenet.links.menu.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+activenet.admin:
+  title: 'ActiveNet settings'
+  parent: system.admin_config_services
+  route_name: activenet.settings
+  weight: 100
+
```
Comment (danylevskyi) [inline modules/custom/activenet/activenet.links.menu.yml]: Let's add new line (as per coding standards).

File: modules/custom/activenet/activenet.routing.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+activenet.settings:
+  path: '/admin/config/services/activenet'
+  defaults:
+    _form: '\Drupal\activenet\Form\SettingsForm'
+    _title: 'ActiveNet settings'
+  requirements:
+    _permission: 'administer activenet'
+
```
Comment (danylevskyi) [inline modules/custom/activenet/activenet.routing.yml]: Again, new line.

**Review by danylevskyi — DISMISSED** [review]
Looks good!

**Comment by mattshoaf** [comment]
New commit removes the debug comment and fixes the new lines.

**Comment by bjwilkins** [comment]
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

**Pull request merged**